mri/operators/fourier/non_cartesian.py

```diff
@@ -378,9 +385,127 @@ def adj_op(self, x):
+    operator: The NUFFTOp object
+        to carry out operation
+    n_coils: int default 1
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition. If n_coils > 1, please organize data as
+            n_coils X data_per_coil
+    """
+    def __init__(self, samples, shape, n_coils=1, density_comp=None,
+                 kernel_width=3, sector_width=8, osf=2, balance_workload=True,
+                 smaps=None):
+        """ Initilize the 'NUFFT' class.
+
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the Fourier domain.
+        shape: tuple of int
+            shape of the image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition
+        density_comp: np.ndarray default None.
+            k-space weighting, density compensation, if not specified
+            equal weightage is given.
+        kernel_width: int default 3
+            interpolation kernel width (usually 3 to 7)
+        sector_width: int default 8
+            sector width to use
+        osf: int default 2
+            oversampling factor (usually between 1 and 2)
+        balance_workload: bool default True
+            whether the workloads need to be balanced
+        smaps: np.ndarray default None
+            Holds the sensitivity maps for SENSE reconstruction
+        """
+        if (n_coils < 1) or (type(n_coils) is not int):
+            raise ValueError('The number of coils should be an integer >= 1')
+        self.n_coils = n_coils
+        self.shape = shape
+        if samples.min() < -0.5 or samples.max() >= 0.5:
+            warnings.warn("Samples will be normalized between [-0.5; 0.5[")
+            samples = normalize_frequency_locations(samples)
+            self.samples = samples
+        if density_comp is None:
+            density_comp = np.ones(samples.shape[0])
+        if smaps is not None:
+            self.uses_smaps = False
+        else:
+            self.uses_smaps = True
+        self.operator = NUFFTOp(
+            np.reshape(samples, samples.shape[::-1], order='F'),
+            shape,
+            n_coils,
+            smaps,
+            density_comp,
+            kernel_width,
+            sector_width,
+            osf,
+            balance_workload
+        )
+
+    def op(self, image):
+        """ This method calculates the masked non-cartesian Fourier transform
+        of a 2D / 3D image.
+
+        Parameters
+        ----------
+        image: np.ndarray
+            input array with the same shape as shape.
+
+        Returns
+        -------
+        np.ndarray
+            Non Uniform Fourier transform of the input image.
+        """
+        if self.n_coils > 1:
+            coeff = self.operator.op(np.asarray(
+                [np.reshape(image_ch.T, image_ch.size) for image_ch in image]
```

Comment:
I don't understand this transpose-reshape. Can you add a comment on it?

Reply:
Yes I was wondering on how to comment on this.
This uses C++ bindings that have a different memory ordering, and this is why we need to do this. DOes this make sense? I will add this now

Reply:
hmm but can you add explicitly the original shape and the end shape. To me it's not clear how we can avoid just reverting to the original image size. There certainly is a mechanism in the `reshape` for that, but it's not explicit.

Reply:
Ideally we would need order='F'
However, I don't think this worked as python seemed to add a layer over the ordering. So ordering 'F' and reshape the way I did seemed to give different results.
This is something I would surely want to explore more, I think I will add it as a TODO

Reply:
ok weird. Does this work even with non-square data?

Reply:
Yup